libs/ledger-state/bench/Memory.hs

```diff
@@ -48,33 +48,36 @@ main = do
               (long "help" <> short 'h' <> help "Display this message.")
         )
         (header "ledger-state:memory - Tool for analyzing memory consumption of ledger state")
-  let cols = [Case, Max, MaxOS, Live, Allocated, GCs]
+  let cols = [Case, Max, MaxOS, Live, Allocated, GCs, WallTime]
   !mEpochStateEntity <- mapM (loadEpochStateEntity . T.pack) (optsSqliteDbFile opts)
   mainWith $ do
     setColumns cols
-    forM_ (optsLedgerStateBinaryFile opts) $ \binFp -> do
-      io "NewEpochState" loadNewEpochState binFp
+    -- forM_ (optsLedgerStateBinaryFile opts) $ \binFp -> do
+    --   io "NewEpochState" loadNewEpochState binFp
     forM_ (optsSqliteDbFile opts) $ \dbFpStr -> do
       let dbFp = T.pack dbFpStr
       forM_ mEpochStateEntity $ \ese ->
         wgroup "EpochState" $ do
+          io "SnapShots (Vector) - no sharing" (loadSnapShotsNoSharingM dbFp) ese
+          io "SnapShots (Vector) - with sharing" (loadSnapShotsWithSharingM dbFp) ese
           io "SnapShots - no sharing" (loadSnapShotsNoSharing dbFp) ese
           io "SnapShots - with sharing" (loadSnapShotsWithSharing dbFp) ese
-      wgroup "Baseline" $ do
-        io "DState" loadDStateNoSharing dbFp
-        io "UTxO" loadUTxONoSharing dbFp
-        io "LedgerState" getLedgerStateNoSharing dbFp
-      wgroup "UTxO (No TxOut)" $ do
-        io "IntMap (KeyMap TxId ())" (loadDbUTxO txIxSharingKeyMap_) dbFp
-        io "KeyMap TxId (IntMap TxId ())" (loadDbUTxO txIdSharingKeyMap_) dbFp
-        io "IntMap (Map TxId ())" (loadDbUTxO txIxSharing_) dbFp
-        io "Map TxIn ()" (loadDbUTxO noSharing_) dbFp
-      wgroup "LedgerState" $ do
-        wgroup "UTxO (Share DState)" $ do
-          io "IntMap (KeyMap TxId TxOut)" getLedgerStateDStateTxIxSharingKeyMap dbFp
-          io "KeyMap TxId (IntMap TxOut)" getLedgerStateDStateTxIdSharingKeyMap dbFp
-          io "IntMap (Map TxId TxOut)" getLedgerStateDStateTxIxSharing dbFp
-          io "Map TxIn TxOut" getLedgerStateDStateSharing dbFp
+
+-- wgroup "Baseline" $ do
+--   io "DState" loadDStateNoSharing dbFp
+--   io "UTxO" loadUTxONoSharing dbFp
+--   io "LedgerState" getLedgerStateNoSharing dbFp
+-- wgroup "UTxO (No TxOut)" $ do
+--   io "IntMap (KeyMap TxId ())" (loadDbUTxO txIxSharingKeyMap_) dbFp
+--   io "KeyMap TxId (IntMap TxId ())" (loadDbUTxO txIdSharingKeyMap_) dbFp
+--   io "IntMap (Map TxId ())" (loadDbUTxO txIxSharing_) dbFp
+--   io "Map TxIn ()" (loadDbUTxO noSharing_) dbFp
+-- wgroup "LedgerState" $ do
+--   wgroup "UTxO (Share DState)" $ do
+--     io "IntMap (KeyMap TxId TxOut)" getLedgerStateDStateTxIxSharingKeyMap dbFp
+--     io "KeyMap TxId (IntMap TxOut)" getLedgerStateDStateTxIdSharingKeyMap dbFp
+--     io "IntMap (Map TxId TxOut)" getLedgerStateDStateTxIxSharing dbFp
+--     io "Map TxIn TxOut" getLedgerStateDStateSharing dbFp
```

Comment:
are these comments helpful to have around? normally I like to not keep such things.

Reply:
These are still useful because these aren't really comments. `weigh` package unfortunately does not have a command line interface that allows selection of which benchmarks to run . Which means the only way right now to select benchmarks is to comment the ones that you don't want to run. I did a simple cli interface that takes some paths, which can be expanded to perform benchmark selection, but the set of benchmarks hasn't really solidified for me to do that. Let's keep that as comment for now.